x509/revocation/ocsp/ocsp.go

```diff
@@ -0,0 +1,422 @@
+package ocsp
+
+import (
+	"crypto"
+	"encoding/asn1"
+	"errors"
+	"math/big"
+	"strconv"
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zcrypto/x509/pkix"
+	"github.com/zmap/zcrypto/x509/revocation/crl"
+)
+
+// Request represents an OCSP request. See RFC 6960.
+type Request struct {
+	HashAlgorithm  crypto.Hash
+	IssuerNameHash []byte
+	IssuerKeyHash  []byte
+	SerialNumber   *big.Int
+}
+
+// https://tools.ietf.org/html/rfc2560#section-4.1.1
+type ocspRequest struct {
```

Comment:
It seems like these would be helpful to have exported -- was there a reason to make them private?

Reply:
No reason in particular to make them private - I just figured it would be a good approach to minimize the number of exported structs. I'll switch them over to public structs though if you think that is more appropriate in this situation.